packages/alfa-rules/test/sia-er62/rule.spec.tsx

```diff
@@ -587,6 +587,480 @@ test(`evaluate() passes an applicable <a> element that removes the default text
+        h.rule.style("span-black", {
+          color: "black",
+        }),
+      ]),
+    ]
+  );
+
+  t.deepEqual(await evaluate(ER62, { document }), [
+    passed(ER62, target, {
+      1: Outcomes.IsDistinguishable([defaultStyle], [hoverStyle], [focusStyle]),
+    }),
+  ]);
+});
+
+test(`evaluate() is applicable to an <a> element with a <p> parent element
+    when the textnode of the <a> element is deeply nested in spans 
+    and has a different foreground color from the <p> element's textnode`, async (t) => {
+  const target = <a href="#">Link</a>;
+
+  const document = h.document(
+    [
+      <p>
+        <span>Hello</span>
+        <span>
+          <span>
+            <span>{target}</span>
+          </span>
+        </span>
+      </p>,
+    ],
+    [
+      h.sheet([
+        h.rule.style("span", {
+          color: "black",
+        }),
+      ]),
+    ]
+  );
+
+  const spanStyle = Err.of(
+    noDistinguishingProperties
+      .withPairings([
+        Contrast.Pairing.of(defaultTextColor, defaultTextColor, 1),
+      ])
+      .withStyle(["color", "rgb(0% 0% 0%)"])
+  );
+
+  t.deepEqual(await evaluate(ER62, { document }), [
+    passed(ER62, target, {
+      1: Outcomes.IsDistinguishable(
+        [defaultStyle, spanStyle],
+        [hoverStyle, spanStyle],
+        [focusStyle, spanStyle]
+      ),
+    }),
+  ]);
+});
+
+test(`evaluate() is applicable to an <a> element with a <p> parent element
+    when the foreground color of the <p> and <a> are the same
+    but the foreground color of the nested <span> in the <a> is different`, async (t) => {
+  const target = (
+    <a href="#">
+      Link <span>world</span>
+    </a>
+  );
+
+  const document = h.document(
+    [
+      <p>
+        Hello
+        {target}
+      </p>,
+    ],
+    [
+      h.sheet([
+        h.rule.style("span", {
+          color: "black",
+        }),
+
+        h.rule.style("p", {
+          color: "#0000EE",
+        }),
+      ]),
+    ]
+  );
+
+  const style = Ok.of(
+    linkProperties.withPairings([
+      Contrast.Pairing.of(defaultLinkColor, defaultLinkColor, 1),
+    ])
+  );
+
+  const spanStyle = noDistinguishingProperties
+    .withPairings([
+      Contrast.Pairing.of(defaultLinkColor, defaultTextColor, 2.23),
+    ])
+    .withStyle(["color", "rgb(0% 0% 0%)"]);
+
+  t.deepEqual(await evaluate(ER62, { document }), [
+    passed(ER62, target, {
+      1: Outcomes.IsDistinguishable(
+        [style, Err.of(spanStyle)],
+        [addCursor(Ok.of(spanStyle)), addCursor(style)],
+        [addOutline(style), Err.of(spanStyle)]
+      ),
+    }),
+  ]);
+});
+
+test(`evaluate() is applicable to an <a> element when there is a <p> parent element
+    with different foreground color`, async (t) => {
+  const target = <a href="#">Link</a>;
+
+  const document = h.document(
+    [
+      <div>
+        <p class="p-one">
+          Hello
+          {target}
+        </p>
+        <p class="p-two">
+          Hello
+          {target}
```

Comment:
We can't use the same element in two different places. Because nodes contain a parent pointer, and `target.parent` is now weird (and `alfa-dom` will actually build something different and not duplicate the element).

These should be two different `target1`, `target2` (and only one being applicable indeed).